x-pack/plugins/security_solution/public/detections/components/alerts_table/default_config.tsx

```diff
@@ -39,30 +39,32 @@ export const buildAlertStatusFilter = (status: Status): Filter[] => [
   },
 ];
 
-export const buildAlertsRuleIdFilter = (ruleId: string): Filter[] => [
-  {
-    meta: {
-      alias: null,
-      negate: false,
-      disabled: false,
-      type: 'phrase',
-      key: 'signal.rule.id',
-      params: {
-        query: ruleId,
-      },
-    },
-    query: {
-      match_phrase: {
-        'signal.rule.id': ruleId,
-      },
-    },
-  },
-];
+export const buildAlertsRuleIdFilter = (ruleId: string | null): Filter[] =>
+  ruleId
+    ? [
+        {
+          meta: {
+            alias: null,
+            negate: false,
+            disabled: false,
+            type: 'phrase',
+            key: 'signal.rule.id',
+            params: {
+              query: ruleId,
+            },
+          },
+          query: {
+            match_phrase: {
+              'signal.rule.id': ruleId,
+            },
+          },
+        },
+      ]
+    : [];
 
-export const buildShowBuildingBlockFilter = (showBuildingBlockAlerts: boolean): Filter[] => [
```

Comment:
I think the logic was inverted here; we were previously returning an empty array if `showBuildingBlockAlerts` was truthy.